kafka/channel/pkg/dispatcher/dispatcher.go

```diff
@@ -167,7 +182,14 @@ func (c consumerMessageHandler) Handle(ctx context.Context, consumerMessage *sar
 		zap.String("topic", consumerMessage.Topic),
 		zap.String("sub", string(c.sub.UID)),
 	)
-	err := c.dispatcher.DispatchMessage(context.Background(), message, nil, destination, reply, deadLetter)
+
+	tracingCtx, span, err := extractTrace(ctx, message)
```

Comment:
I would prefer to reuse ctx whenever updating context, otherwise it is easy to accidentally use the wrong context.